compaction/compaction_manager.hh

```diff
@@ -160,6 +160,8 @@ private:
     std::optional<utils::observer<uint32_t>> _throughput_option_observer;
     serialized_action _update_compaction_static_shares_action;
     utils::observer<float> _compaction_static_shares_observer;
+    serialized_action _update_compaction_max_shares_action;
```

Comment:
I see that the static shares config also has this, but would still like to understand why this is needed? Isn't it enough to propagate the updateable value down to the backlog manager, which would pull the most fresh value on each `update_controller()` call?

Reply:
You're right - just updating controller directly is enough. I misunderstood how `serialized_action` works. It is required only if  the action is asynchronous, which is not the case here. I have removed it and I have updated `update_max_shares` to return void. Looks like _update_compaction_static_shares_action can also be removed as well but I have not made that change in this PR.

Reply:
Yes, fixing `_update_compaction_static_shares_action` should be another PR.